Cython/Utility/StringTools.c

```diff
@@ -610,28 +610,71 @@ static CYTHON_INLINE PyObject* __Pyx_PyUnicode_Substring(
 // Py_UNICODE_ISTITLE() doesn't match unicode.istitle() as the latter
 // additionally allows character that comply with Py_UNICODE_ISUPPER()
 
+#if CYTHON_COMPILING_IN_LIMITED_API
+static int __Pyx_Py_UNICODE_ISTITLE(Py_UCS4 uchar); /* proto */
+#else
 static CYTHON_INLINE int __Pyx_Py_UNICODE_ISTITLE(Py_UCS4 uchar) {
     return Py_UNICODE_ISTITLE(uchar) || Py_UNICODE_ISUPPER(uchar);
 }
+#endif
+
+/////////////// py_unicode_istitle /////////////////
+
+#if CYTHON_COMPILING_IN_LIMITED_API
+static int __Pyx_Py_UNICODE_ISTITLE(Py_UCS4 uchar) {
+    int result;
+    PyObject *py_result, *ustring;
+    // Fast path for ascii
+    if (uchar < 128) {
```

Comment:
First non-ASCII title character in Unicode is 192 (`"À"`), so we could go further:
```suggestion
    if (uchar < 192) {
```